### DISCUSSION TITLE
Margin shorthand

### DISCUSSION THREAD

**Discussion by dgwyer** [observablehq/plot#1682] [Q&A] [answered]
In the [layout options doc](https://observablehq.com/plot/features/plots#layout-options) it states that you can use `margin` as a shorthand for specifying `marginLeft`, `marginRight`, `marginTop`, and `marginBottom` individually. But there is no example usage that I can see.

Is this shorthand an object, or an array perhaps?

Array
```
Plot.plot({
  margin: [20, 20 , 20, 20],
  marks: [ ... ]
})
```

Object v1
```
Plot.plot({
  margin: {
    marginTop: 20,
    marginBottom: 20,
    marginLeft: 20,
    marginRight: 20,
  },
  marks: [ ... ]
})
```

Object v2
```
Plot.plot({
  margin: {
    top: 20,
    bottom: 20,
    left: 20,
    right: 20,
  },
  marks: [ ... ]
})
```

I've tried these variations but they don't seem to work. I don't know if it's my implementation or not. Can you confirm usage, please?

**Comment by mbostock** [comment, marked as answer]
It’s shorthand for specifying all four margins as *the same value*. For example, `margin: 20`.

```
Plot.plot({
  margin: 20,
  marks: [ ... ]
})
```

is equivalent to

```
Plot.plot({
  marginTop: 20,
  marginBottom: 20,
  marginLeft: 20,
  marginRight: 20,
  marks: [ ... ]
})
```

**Reply by dgwyer** [reply]
Oh, so only if they're all the same. Thanks for clarifying.
